packages/core/types/index.js

```diff
@@ -1931,3 +1976,31 @@ export interface IDisposable {
 export type AsyncSubscription = {|
   unsubscribe(): Promise<mixed>,
 |};
+
+export interface PluginApplicationProfiler {
+  /** Returns whether the application profiler is enabled. Use this to avoid possibly expensive calculations
+   * of arguments to `createMeasurement` - for example if you need to determine the entry of a bundle to pass it
+   * in as the <code>argumentName</code>, you would only do this if the application profiler is enabled.
+   */
+  get enabled(): boolean;
```

Comment:
Nit: for all the other public types, we've been using a readonly property (so `+enabled: boolean;`). For example `bundle.needsStableName` is in reality a getter in `core/src/public/Bundle.js`